### PR TITLE
add constraints and namelist

### PR DESCRIPTION
**Pull request by RuiyuSun**
## Description

(Instructions: this, and all subsequent sections of text should be removed and filled in as appropriate.)
Provide a detailed description of what this PR does.  
What bug does it fix, or what feature does it add?  
Is a change of answers expected from this PR?  

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

How were these changes tested?  
What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>

### CODE REVIEW

**Review by climbfuji — COMMENTED** [review]
This PR does not point to the correct ccpp-physics submodule hash, it still points to 990b9d0416e340e40e4b136561a85f63d46671e2 (from NCAR ccpp-physics master/main, "Merge pull request #589 from XuLi-NOAA/feature/ufs-nst"). There is also no `.gitmodules` update.

File: ccpp/data/GFS_typedefs.F90 [hunk 1]
```diff
@@ -762,8 +762,12 @@ module GFS_typedefs
     !--- Thompson's microphysical parameters
     logical              :: ltaerosol       !< flag for aerosol version
     logical              :: lradar          !< flag for radar reflectivity
+    logical              :: fcons           !< flag for constraints
+    logical              :: ffixneg         !< flag for fixing negative traces
     real(kind=kind_phys) :: nsradar_reset   !< seconds between resetting radar reflectivity calculation
     real(kind=kind_phys) :: ttendlim        !< temperature tendency limiter per time step in K/s
+    real(kind=kind_phys) :: crtclw          !< ritical value for cloud water to rain in kg kg-1
```
Comment (climbfuji) [inline ccpp/data/GFS_typedefs.F90:769]: This should be "critical", not "ritical"

**Review by climbfuji — CHANGES_REQUESTED** [review]
The new namelist options have somewhat cryptic names. I would also invite @grantfirl and/or @ligiabernardet to look at the proposed standard names. There is a typo in one comment that should be fixed.

File: ccpp/data/GFS_typedefs.F90 (hunk 1 above)
Comment (RuiyuSun) [reply ccpp/data/GFS_typedefs.F90:769]: yes. It should be "critical".

**Comment by climbfuji** [comment]
Closed as agreed upon in the discussion in the ccpp-physics PR.

**Pull request closed without merging**